src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,701 @@
+
+  if (!this->dataPtr->rgbCamera)
+  {
+    // Create rendering camera
+    this->dataPtr->boundingboxCamera =
+      this->Scene()->CreateBoundingBoxCamera(this->Name());
+
+    this->dataPtr->rgbCamera = this->Scene()->CreateCamera(
+      this->Name() + "_rgbCamera");
+  }
+
+  auto width = sdfCamera->ImageWidth();
+  auto height = sdfCamera->ImageHeight();
+
+  // Set Camera Properties
+  this->dataPtr->rgbCamera->SetImageFormat(rendering::PF_R8G8B8);
+  this->dataPtr->rgbCamera->SetImageWidth(width);
+  this->dataPtr->rgbCamera->SetImageHeight(height);
+  this->dataPtr->rgbCamera->SetVisibilityMask(sdfCamera->VisibilityMask());
+  this->dataPtr->rgbCamera->SetNearClipPlane(sdfCamera->NearClip());
+  this->dataPtr->rgbCamera->SetFarClipPlane(sdfCamera->FarClip());
+  math::Angle angle = sdfCamera->HorizontalFov();
+  if (angle < 0.01 || angle > IGN_PI*2)
+  {
+    ignerr << "Invalid horizontal field of view [" << angle << "]\n";
+    return false;
+  }
+  double aspectRatio = static_cast<double>(width)/height;
+  this->dataPtr->rgbCamera->SetAspectRatio(aspectRatio);
+  this->dataPtr->rgbCamera->SetHFOV(angle);
+
+  this->dataPtr->boundingboxCamera->SetImageWidth(width);
+  this->dataPtr->boundingboxCamera->SetImageHeight(height);
+  this->dataPtr->boundingboxCamera->SetNearClipPlane(sdfCamera->NearClip());
+  this->dataPtr->boundingboxCamera->SetFarClipPlane(sdfCamera->FarClip());
+  this->dataPtr->boundingboxCamera->SetImageFormat(
+    rendering::PixelFormat::PF_R8G8B8);
+  this->dataPtr->boundingboxCamera->SetAspectRatio(aspectRatio);
+  this->dataPtr->boundingboxCamera->SetHFOV(angle);
+  this->dataPtr->boundingboxCamera->SetVisibilityMask(
+    sdfCamera->VisibilityMask());
+  this->dataPtr->boundingboxCamera->SetBoundingBoxType(this->dataPtr->type);
+
+  // Add the camera to the scene
+  this->Scene()->RootVisual()->AddChild(this->dataPtr->rgbCamera);
+  this->Scene()->RootVisual()->AddChild(this->dataPtr->boundingboxCamera);
+
+  // Add the rendering sensors to handle its render
+  this->AddSensor(this->dataPtr->boundingboxCamera);
+  this->AddSensor(this->dataPtr->rgbCamera);
+
+  // Create the directory to store frames
+  if (sdfCamera->SaveFrames())
+  {
+    this->dataPtr->savePath = sdfCamera->SaveFramesPath();
+    this->dataPtr->saveImageFolder =
+      this->dataPtr->savePath + this->dataPtr->saveImageFolder;
+    this->dataPtr->saveBoxesFolder =
+      this->dataPtr->savePath + this->dataPtr->saveBoxesFolder;
+    this->dataPtr->saveSample = true;
+
+    // Set the save counter to be equal number of images in the folder + 1
+    // to continue adding to the images in the folder (multi scene datasets)
+    if (ignition::common::isDirectory(this->dataPtr->saveImageFolder))
+    {
+      for (const auto &entry : std::filesystem::directory_iterator(
+        this->dataPtr->saveImageFolder))
+      {
+        // To disable unused warning
+        (void)entry;
+        this->dataPtr->saveCounter++;
+      }
+    }
+  }
```

Comment:
As mentioned in https://github.com/ignitionrobotics/ign-sensors/pull/133#issuecomment-910513149, can you change this to use `ignition::common::DirIter` so that we don't have compilation issues?